src/OmniSharp.Roslyn.CSharp/Helpers/QuickFixExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using OmniSharp.Models;
+
+namespace OmniSharp.Helpers
+{
+    internal static class QuickFixExtensions
+    {
+        internal static async Task AddQuickFix(this ICollection<QuickFix> quickFixes, OmniSharpWorkspace workspace, Location location)
+        {
+            if (location.IsInSource)
+            {
+                var quickFix = await QuickFixHelper.GetQuickFix(workspace, location);
```

Comment:
I hadn't looked at this code before, so forgive me if I'm creating work for you. :smile: However, none of these helpers should be async. Ultimately, these call into `QuickFixHelper`, which is async just to ask for the tree and then the `SourceText`. This is all remarkably inefficient.

Instead of passing in `OmniSharpWorkspace` could these just take `SourceText`? After all, we already have (or nearly have) a `SourceText` in all of the places that call into `QuickFixHelper.GetQuickFix(...)`.